### PR TITLE
Add logic for selecting the right version of pgort140

### PR DESCRIPTION
**Pull request by DrewScoggins**
This fixes an issue where the Arm64 PGO instrumented SDK had the x64 version of pgort140.dll

### CODE REVIEW

File: src/coreclr/jit/CMakeLists.txt [hunk 1]
```diff
@@ -626,7 +626,11 @@ endif()
 if (CLR_CMAKE_TARGET_WIN32 AND CLR_CMAKE_PGO_INSTRUMENT)
   # Copy PGO dependency to target dir
   set(PGORT_DLL "pgort140.dll")
-  find_path(PGORT_DIR ${PGORT_DLL} REQUIRED)
+  if(ARCH_TARGET_NAME STREQUAL "arm64")
```
Comment (am11) [inline src/coreclr/jit/CMakeLists.txt:629]: nit: quotes are not needed (see line 617). Condition can be simplified to:
```suggestion
  if(CLR_CMAKE_TARGET_ARCH_ARM64)
```

**Comment by DrewScoggins** [comment]
/azprun "CoreCLR Product Build windows arm64 release PGO"

**Comment by DrewScoggins** [comment]
I have already tested this on the PGO builds locally, so if all the other checks pass I would like to merge this ASAP. @agocke @dotnet/runtime-infrastructure

File: src/coreclr/jit/CMakeLists.txt [hunk 2]
```diff
@@ -626,7 +626,11 @@ endif()
 if (CLR_CMAKE_TARGET_WIN32 AND CLR_CMAKE_PGO_INSTRUMENT)
   # Copy PGO dependency to target dir
   set(PGORT_DLL "pgort140.dll")
-  find_path(PGORT_DIR ${PGORT_DLL} REQUIRED)
+  if(CLR_CMAKE_TARGET_ARCH_ARM64)
+    cmake_path(SET PGORT_DIR "$ENV{VCToolsInstallDir}bin/arm64")
+  else()
```
Comment (agocke) [inline src/coreclr/jit/CMakeLists.txt:631]: I think you need an extra check here for x86

Comment (am11) [reply src/coreclr/jit/CMakeLists.txt:631]: Another option is to keep using `find_path` and leverage `HINTS` options https://cmake.org/cmake/help/latest/command/find_path.html (if the original intent was to disambiguate ${PGORT_DLL} being located)

Comment (DrewScoggins) [reply src/coreclr/jit/CMakeLists.txt:631]: If we use the `HINTS` option then we would have to guarantee that we do not ever have the Arm64 tooling installed on the build machine, or we will always find the Arm64 version, due to the order that `find_path` searchs.

Comment (DrewScoggins) [reply src/coreclr/jit/CMakeLists.txt:631]: I don't know if that is true, but even if it is I think I would prefer to be more explicit. That way we don't get broken by someone changing that invariant.

File: src/coreclr/jit/CMakeLists.txt [hunk 3]
```diff
@@ -626,7 +626,14 @@ endif()
 if (CLR_CMAKE_TARGET_WIN32 AND CLR_CMAKE_PGO_INSTRUMENT)
   # Copy PGO dependency to target dir
   set(PGORT_DLL "pgort140.dll")
-  find_path(PGORT_DIR ${PGORT_DLL} REQUIRED)
+  cmake_path(SET PGORT_DIR_ARM64 "$ENV{VCToolsInstallDir}bin/arm64")
```
Comment (agocke) [inline src/coreclr/jit/CMakeLists.txt:629]: I think this first assignment is dead code.

**Review by agocke — APPROVED** [review]
LGTM aside from what looks like dead code

File: src/coreclr/jit/CMakeLists.txt (hunk 3 above)
Comment (DrewScoggins) [reply src/coreclr/jit/CMakeLists.txt:629]: Yeah, that was from the hint path experiment. I will remove that.

File: src/coreclr/jit/CMakeLists.txt [hunk 4]
```diff
@@ -626,7 +626,14 @@ endif()
...
+  if(CLR_CMAKE_TARGET_ARCH_ARM64)
+    cmake_path(SET PGORT_DIR "$ENV{VCToolsInstallDir}bin/arm64")
+  elseif(CLR_CMAKE_TARGET_ARCH_I386)
+    cmake_path(SET PGORT_DIR "$ENV{VCToolsInstallDir}bin/Hostx86/x86")
```
Comment (agocke) [inline src/coreclr/jit/CMakeLists.txt:633]: This should probably work, but I think we only support building on x64, so technically this should be `Hostx64/x86`

Comment (DrewScoggins) [reply src/coreclr/jit/CMakeLists.txt:632]: I will fix it up. Want to do it right :)

Comment (DrewScoggins) [reply src/coreclr/jit/CMakeLists.txt:632]: Actually, `Hostx64/x86` does not contain `pgort140.dll`. So I am going to stick with what I have.

**Comment by mrsharm** [comment]
We detected that might be caused by this PR via the August Perf Report and looks like it was reported in a few places but hasn't been triaged yet: 

- https://github.com/dotnet/perf-autofiling-issues/issues/6936
- https://github.com/dotnet/perf-autofiling-issues/issues/6931

Here are the details from the perf report: 

<details> 

System.Numerics.Tests.Perf_BigInteger.Parse(numberString: 12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890)

| Result |Ratio | Alloc Delta | Operating System      | Bit   | Processor Name                                  |
| ------ |-----:| -----------:| --------------------- | ----- | ----------------------------------------------- |
| Slower | 0.72 |          +0 | Windows 11            | Arm64 | Microsoft SQ1 3.0 GHz                           |
| Slower | 0.72 |          +0 | Windows 11            | Arm64 | Microsoft SQ1 3.0 GHz                           |
| Slower | 0.77 |          +0 | macOS Monterey 12.3   | Arm64 | Apple M1 Max                                    |
| Same   | 0.92 |          +0 | Windows 10            | X64   | Intel Xeon CPU E5-1650 v4 3.60GHz               |
| Same   | 0.92 |          +0 | Windows 10            | X64   | Intel Core i7-6700 CPU 3.40GHz (Skylake)        |
| Same   | 0.90 |          +0 | Windows 10            | X64   | Intel Core i7-6700 CPU 3.40GHz (Skylake)        |
| Same   | 0.90 |          +0 | Windows 10            | X64   | Intel Core i7-8650U CPU 1.90GHz (Kaby Lake R)   |
| Same   | 0.93 |          +0 | Windows 10            | X64   | Intel Core i9-10900K CPU 3.70GHz                |
| Slower | 0.85 |          +0 | Windows 11            | X64   | AMD Ryzen Threadripper PRO 3945WX 12-Cores      |
| Slower | 0.88 |          +0 | Windows 11            | X64   | AMD Ryzen 9 3950X                               |
| Slower | 0.89 |          +0 | Windows 11            | X64   | AMD Ryzen 9 5900X                               |
| Same   | 0.90 |          +0 | Windows 11            | X64   | AMD Ryzen 9 5950X                               |
| Slower | 0.89 |          +0 | Windows 11            | X64   | Intel Core i7-8700 CPU 3.20GHz (Coffee Lake)    |
| Slower | 0.89 |          +0 | Windows 11            | X64   | Intel Core i9-10900K CPU 3.70GHz                |
| Same   | 0.93 |          +0 | Windows 11            | X64   | 11th Gen Intel Core i9-11900H 2.50GHz           |
| Same   | 1.01 |          +0 | ubuntu 18.04          | X64   | Intel Xeon CPU E5-1650 v4 3.60GHz               |
| Same   | 0.99 |          +0 | ubuntu 18.04          | X64   | Intel Core i7-2720QM CPU 2.20GHz (Sandy Bridge) |
| Slower | 0.84 |          +0 | ubuntu 18.04          | X64   | Intel Core i7-8700 CPU 3.20GHz (Coffee Lake)    |
| Slower | 0.86 |          +0 | ubuntu 20.04          | X64   | AMD Ryzen 9 5900X                               |
| Slower | 0.80 |          +0 | ubuntu 20.04          | X64   | Intel Core i9-10900K CPU 3.70GHz                |
| Slower | 0.85 |          +0 | Windows 10            | X86   | Intel Xeon CPU E5-1650 v4 3.60GHz               |
| Slower | 0.85 |          +0 | Windows 10            | X86   | Intel Core i7-6700 CPU 3.40GHz (Skylake)        |
| Slower | 0.82 |          +0 | Windows 11            | X86   | AMD Ryzen Threadripper PRO 3945WX 12-Cores      |
| Slower | 0.83 |          +0 | macOS Big Sur 11.6.8  | X64   | Intel Core i5-4278U CPU 2.60GHz (Haswell)       |
| Slower | 0.77 |          +0 | macOS Monterey 12.3.1 | X64   | Intel Core i7-5557U CPU 3.10GHz (Broadwell)     |
| Slower | 0.86 |          +0 | macOS Monterey 12.4   | X64   | Intel Core i5-4278U CPU 2.60GHz (Haswell)       |


</details>

**Comment by DrewScoggins** [comment]
I doubt that this change could have caused this regression. It does not change anything about the shipping product, it just changes one of the binaries that gets bin-placed for the instrumented SDK.

**Pull request merged**